Partial configuration for Digital Ocean-based config servers

aranhoide opened the pull request:
As of this writing, the uberjar itself needs to be manually built and uploaded to the cloudmaster, then you need to call highstate on the cloudmaster itself before launching/updating config servers.

Launch a config server by calling `launch_config_server.py` as root.  After that, you'll need to manually add it to the Cloudflare round robin.

This is as much automation as made sense for a first shot at launching some 2-8 config servers.

aranhoide commented on salt/cloudmaster/init.sls:
```diff
@@ -3,6 +3,12 @@ include:
+# config-server is in a private repo.
+update-config-server-uberjar:
+  cmd.run:
+    - name: "cp -f /home/lantern/config-server.jar /srv/salt/config_server/config-server.jar"
```
I manually `scp` the uberjars to `donyc3`, which is the only cloudmaster that will be launching config servers.  I've run `./inallcms ./ssh_cloudmaster.py 'sudo test -e /home/lantern/config-server.jar || sudo touch /home/lantern/config-server.jar'` to put bogus files in all of them so this doesn't raise an (otherwise harmless) false alarm.

aranhoide commented on salt/cloudmaster/launch_config_server.py:
```diff
@@ -0,0 +1,21 @@
+
+
+def launch_config_server():
+    # XXX: this doesn't perform a complete setup yet.
```
It's pretty much complete actually, except (i) the uberjar must be already in the cloudmaster, which must have been done manually, and (ii) adding the A entries to Cloudflare is manual ATM.

aranhoide commented on salt/config_server/config-server.conf:
```diff
@@ -0,0 +1,11 @@
+
+chdir /home/lantern
+
+exec su lantern -c "java -jar /home/lantern/config-server.jar 2>&1 | logger -t config-server" 2>&1 | logger -t config-server
```
The `su lantern` part is only so we get the environment variables for this user without having to explicitly list them in this config file.

aranhoide commented:
@oxtoacart thanks for grabbing this one!  The most sensitive part of this PR would be the refactoring of existing vps_util (and, to some extent, refill_srvq) functions.  I have tested these in isolation but I haven't launched a non-config-server machine yet.  I'll do so in an hour or so.

oxtoacart commented:
@aranhoide FYI, this has merge conflicts now (probably because you helped me with hussh earlier today).

aranhoide commented:
@oxtoacart thanks for the heads up!  I'll resolve them, finally launch a test proxy, and push again.

aranhoide commented:
@oxtoacart : I've fixed the merge conflict and tested launching a cloudmaster.  I've also fixed the minor annoyance that a state would fail unless the cloudmaster had a config server uberjar (or something with the same file name).

aranhoide commented:
Merging since I'm actually going to deploy this to donyc3 to launch a few config servers.  This is a singularly good time to do so, in the lowest of the day's activity and with the upcoming weekend.

oxtoacart commented:
@aranhoide Sorry I didn't get around to this.  I don't mean to be a bottleneck.

aranhoide commented:
No worries, you are not.  I didn't deploy this before because I was doing other checks, not because I was waiting on this PR.

BTW, this stuff is live now. :)

aranhoide commented:
@oxtoacart ^
